sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -0,0 +1,283 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Exposes client options for logging within a <see cref="ClientPipeline"/>.
+/// </summary>
+public class ClientLoggingOptions
+{
+    private bool _frozen;
+
+    private bool? _enableLogging;
+    private bool? _enableMessageLogging;
+    private bool? _enableMessageContentLogging;
+    private int? _messageContentSizeLimit;
+    private ILoggerFactory? _loggerFactory;
+    private PipelineMessageSanitizer? _sanitizer;
+    private ChangeTrackingStringList? _allowedHeaderNames;
+    private ChangeTrackingStringList? _allowedQueryParameters;
+
+    private static HashSet<string> DefaultAllowedHeaderNames { get; } = [
```

Comment:
Do we want to `readonly` these or do we want to allow the reference to be reassigned?